packages/graphql/lib/schema-builder/collections/array.collection.ts

```diff
@@ -0,0 +1,35 @@
+export class ArrayCollection<T> {
+  private array: T[] = [];
+
+  constructor(private globalArray: Array<T>) {}
+
+  getAll() {
+    return this.array;
+  }
+
+  push(...items): number {
+    this.globalArray.push(...items);
+    return this.array.push(...items);
+  }
+
+  unshift(...items): number {
+    this.globalArray.unshift(...items);
+    return this.array.unshift(...items);
+  }
+
+  reverse() {
+    return this.array.reverse();
+  }
+
+  reduce<U>(
+    callbackfn: (
+      previousValue: U,
+      currentValue: T,
+      currentIndex: number,
+      array: T[],
+    ) => U,
+    initialValue: U,
+  ): U {
+    return this.array.reduce(callbackfn, initialValue);
+  }
```

Comment:
We should remove `reverse` and `reduce` and let users grab the underlying array instance instead (and call these methods directly). For this, `getAll()` method can be used (which should be renamed as well as it's somewhat confusing atm)